src/python/pants/core/goals/check_test.py

```diff
@@ -32,12 +26,12 @@ class MockTarget(Target):
     core_fields = (Sources,)
 
 
-class MockTypecheckFieldSet(FieldSet):
+class MockCheckFieldSet(FieldSet):
     required_fields = (Sources,)
 
 
-class MockTypecheckRequest(TypecheckRequest, metaclass=ABCMeta):
-    field_set_type = MockTypecheckFieldSet
+class MockCheckRequest(CheckRequest, metaclass=ABCMeta):
+    field_set_type = MockCheckFieldSet
     typechecker_name: ClassVar[str]
```

Comment:
Should probably be called `checker_name`. There are a couple other uses of "typecheck" if you grep.

Reply:
Yea, there are a lot. This PR is backwards compatible currently: you get a warning on import of the old module, but otherwise things continue to work. Changing field names, on the other hand, would be breaking.

Do we want to do that here? I guess that I was lazy.

Reply:
Given that it's a test, yeah I think it's helpful to update.